ansible/smoke.yml

```diff
@@ -150,3 +150,28 @@
       become: false
       run_once: true
       when: smoke_enable_cdn_checks | default(True)
+
+- name: catalog-next.data.gov /dataset redirection
+  hosts: all
+  gather_facts: false
+  tasks:
+    - name: assert only one redirection to https and /dataset
+      vars:
+        url: http://catalog-next.data.gov
+      block:
+        - name: look for expected redirection
+          uri:
+            url: "{{ url }}"
+            status_code: 302
+            follow_redirects: all
+            maximum_redirection: 1
+            register: response
+        - name: assert redirection
+          assert:
+            that: response.url == 'https://catalog-next.data.gov/dataset'
+            fail_msg: |
+              We expect a redirection to HTTPS and to /dataset
+      delegate_to: localhost
+      become: false
+      run_once: true
+      when: ckan_uses_gunicorn | default(True)
```

Comment:
Why ckan_uses_gunicorn? ckan_uses_gunicorn variable is not defined for all hosts, so it will likely always end up with `default(True)`.

Reply:
Since this task is delegated to `localhost` and `run_once` is present, the `when` line is not needed, I just delete it.